sympy/core/numbers.py

```diff
@@ -161,7 +161,7 @@ def igcd(*args):
             a = _gcdcache[(a, b)]
         except KeyError:
             a, b = as_int(a), as_int(b)
-
+            t = (a, b)
```

Comment:
`a, b` are modified in the lines below and when code reaches `_gcdcache`, `b` is 0 and therefore cache only `(a, 0) -> a` which is a trivial result.